plugins/uefi/efi/fwup-efi.c

```diff
@@ -17,16 +17,28 @@ fwup_delete_variable(CHAR16 *name, EFI_GUID *guid)
 {
 	EFI_STATUS rc;
 	UINT32 attrs = 0;
+	UINTN size = 0;
+	_cleanup_free VOID *buf = NULL;
 
 	/* get the attrs so we can delete it */
-	rc = uefi_call_wrapper(RT->GetVariable, 5, name, guid, &attrs, NULL, NULL);
+	rc = uefi_call_wrapper(RT->GetVariable, 5, name, guid, &attrs, &size, NULL);
```

Comment:
Is this working around a BIOS bug somewhere? I'm pretty sure NULL works here...

Reply:
If you have UEFI spec 2.8 you can check the GetVariable returned, p.233.
When DataSize is NULL, UEFI firmeare should return EFI_INVALID_PARAMETER. Means that DataSize cannot be NULL.

Reply:
Okay, thanks for checking.